rtc_tenclave/src/crypto.rs

```diff
@@ -137,17 +185,34 @@ impl RtcCrypto for SodaBoxCrypto {
         let nonce = self.get_nonce()?;
         // Length is padded here since the message needs to be padded with 32 `0_u8`
         // at the front
-        let mut ciphertext = vec![0_u8; message.expose_secret().len() + 32].into_boxed_slice();
-
+        let mut ciphertext = vec![0_u8; message.expose_secret().len() + CRYPTO_BOX_ZEROBYTES];
+
+        // Docs: https://nacl.cr.yp.to/box.html
+        //
+        // "The caller must ensure, before calling the C NaCl crypto_box function,
+        // that the first crypto_box_ZEROBYTES bytes of the message m are all 0.
+        // The crypto_box function ensures that the first crypto_box_BOXZEROBYTES
+        // bytes of the ciphertext c are all 0."
+        //
         match sodalite::box_(
             &mut ciphertext,
-            // Need to pad with 32 0s see https://nacl.cr.yp.to/secretbox.html
-            &[&[0u8; 32] as &[u8], message.expose_secret()].concat(),
+            &[
+                &[0u8; CRYPTO_BOX_ZEROBYTES] as &[u8],
+                message.expose_secret(),
+            ]
+            .concat(),
             &nonce,
             their_pk,
             self.private_key.expose_secret(),
         ) {
-            Ok(_) => Ok(EncryptedMessage { ciphertext, nonce }),
+            Ok(_) => {
+                ciphertext.rotate_left(CRYPTO_BOX_BOXZEROBYTES);
+                ciphertext.truncate(CRYPTO_BOX_BOXZEROBYTES);
```

Comment:
This should probably be the padded length minus the padding?

```suggestion
                ciphertext.truncate(ciphertext.len() - CRYPTO_BOX_BOXZEROBYTES);
```

Reply:
Thanks for catching this!